larreco/RecoAlg/PMAlg/PmaHit3D.h

```diff
@@ -0,0 +1,104 @@
+/**
+ *  @file   PmaHit3D.h
+ *
+ *  @author D.Stefan and R.Sulej
+ * 
+ *  @brief  Implementation of the Projection Matching Algorithm
+ *
+ *          Hit 3D wrapped around recob::Hit. Adds support for PMA optimizations.
+ *          See PmaTrack3D.h file for details.
+ */
+
+#ifndef PmaHit3D_h
+#define PmaHit3D_h
+
+#include "lardata/RecoBase/Hit.h"
+
+#include "larcore/Geometry/Geometry.h"
+
+#include <functional>
+
+#include "TVector2.h"
+#include "TVector3.h"
+
+namespace pma
+{
+	class Hit3D;
+	struct bTrajectory3DOrderLess;
+
+	class Track3D;
+}
+
+class pma::Hit3D
+{
+	friend class Track3D;
+	friend struct bTrajectory3DOrderLess;
+
+public:
+	Hit3D(void);
+	Hit3D(art::Ptr< recob::Hit > src);
+	Hit3D(unsigned int wire, unsigned int view, unsigned int tpc, unsigned int cryo,
+		float peaktime, float ampl, float area);
+	Hit3D(const pma::Hit3D& src);
+	virtual ~Hit3D(void) {}
+
+	art::Ptr< recob::Hit > Hit2DPtr(void) const { return fHit; }
+
+	TVector3 const & Point3D(void) const { return fPoint3D; }
+
+	void SetPoint3D(const TVector3& p3d) { fPoint3D = p3d; }
+
+	TVector2 const & Point2D(void) const { return fPoint2D; }
+	TVector2 const & Projection2D(void) const { return fProjection2D; }
+
+	unsigned int Cryo(void) const { return fCryo; }
+	unsigned int TPC(void) const { return fTPC; }
+	unsigned int View2D(void) const { return fPlane; }
```

Comment:
From the code, is it correct to say that the only valid values are `geo::Z`, `geo::U` and `geo::V`? If so, how about changing this to return an enum?